src/CFamily/CMake/MacroEvaluationService.cs

```diff
@@ -34,17 +40,62 @@ internal interface IMacroEvaluationService
 
     internal class MacroEvaluationService : IMacroEvaluationService
     {
+        private readonly ILogger logger;
+        private readonly IMacroEvaluator macroEvaluator;
+
+        // Looks for patterns like "${name}" and "${prefix.name}" and captures the [prefix, name] tuple.
+        private static readonly Regex CMakeSettingsMacroRegex = new Regex(
+            "\\${" +                 // Match opening literals "${". $ is a special Regex symbol that needs be escaped,
+                                     // and "\" needs to be escaped in the C# string.
+            "((?<prefix>\\w+)\\.)" + // One or more word chars followed by ".". Like "$" above, the "." needs to be escaped with "\\".
+                                     // The word chars are captured in a group called "prefix".
+            "?" +                    // Match the previous group 0 or 1 times i.e. the prefix is optional.
+            "(?<name>\\w+)" +        // Capture one or more word consecutive word chars in a group called "name".
+            "}");                    // Match the closing literal.
+
+        public MacroEvaluationService(ILogger logger)
+            :this(logger, new MacroEvaluator()) {}
+
+        internal MacroEvaluationService(ILogger logger, IMacroEvaluator macroEvaluator)
+        {
+            this.logger = logger;
+            this.macroEvaluator = macroEvaluator;
+        }
+
         public string Evaluate(string input, string activeConfiguration, string workspaceRootDir)
         {
             if (input == null)
             {
                 return null;
             }
+
```

Comment:
The service now orchestrates the process of finding the macros, asking for the values, and replacing them in the input.

The actual calculation of individual values is delegated to the new `IMacroEvaluator`.